### PR TITLE
Upgrade dependencies and squash (most) reflection warnings

### PR DESCRIPTION
**Pull request by pmonks**
Apologies for the fly-by PR, but I use `lein check` a lot and so when a library I'm using has a bunch of reflection warnings it makes it harder to improve my own code.

Tests pass for me here:
```
$ lein do check, test
Compiling namespace clj-jgit.util
Compiling namespace clj-jgit.porcelain
Reflection warning, clj_jgit/porcelain.clj:549:18 - call to method getJSch can't be resolved (target class is unknown).
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Compiling namespace clj-jgit.querying
Compiling namespace clj-jgit.internal
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

lein test clj-jgit.test.core

lein test clj-jgit.test.helpers

lein test clj-jgit.test.internal

lein test clj-jgit.test.porcelain

lein test clj-jgit.test.querying

lein test clj-jgit.test.util

Ran 12 tests containing 69 assertions.
0 failures, 0 errors.
```

Note that I'm new to JGit, so please carefully review all of the type hints I added to make sure they're correct!

### CODE REVIEW

**Comment by edannenberg** [comment]
Thanks, will probably gonna be a few days though before I have time for a proper review. As for the Travis setup, see #56 for current status. Should have a green lamp soonish again.

**Comment by pmonks** [comment]
Thanks @edannenberg - no hurry from my end; I'd rather you were thorough in your review rather than rushing this in.

And sorry for the mess with `.travis.yml` - it looks like [TravisCI is in a bit of a pickle wrt JVM versions at the moment](https://github.com/travis-ci/travis-ci/issues/8751), and as a result I wasn't able to get all versions and "makes" (Open JDK vs Oracle JDK) of JVM testable at the same time.  My last commit (take 3) was the best alternative I could find in the limited time I had available, and even then JDK 6 is not being tested (you'll notice JDK6 is being installed _after_ the `jdk_switcher` command is run - there may be a clever way to force TravisCI to rerun that post-installation but I didn't find anything with a quick google).  Feel free to revert those changes as they're unrelated to my original goal (removing reflection warnings).

**Comment by pmonks** [comment]
Ha looks like the JVM version issue may have been "fixed" for us by the JGit team - the latest version of JGit is [compiled for Java 1.8](https://travis-ci.org/clj-jgit/clj-jgit/jobs/358681290#L592)!

**Comment by jaromil** [comment]
utACK, reads good to me. Will test regression in my own software soon, but if others do earlier then I have nothing against merging really.

**Comment by edannenberg** [comment]
The fixed `.travis.yml` is merged, feel free to rebase the PR, no worries though if you are busy. The failed builds above are me not realising that the tests currently only work in master branch, which is kinda suboptimal and should get refactored.

**Comment by pmonks** [comment]
Done - thanks @edannenberg !

File: src/clj_jgit/porcelain.clj [hunk 1]
```diff
@@ -787,11 +787,11 @@
...
          blame-result)))
 
-(defn get-blob-id
-  [repo commit path]
+(defn ^org.eclipse.jgit.lib.ObjectId get-blob-id
+  [^Git repo ^org.eclipse.jgit.revwalk.RevCommit commit ^String path]
   (let [tree-walk (TreeWalk/forPath (.getRepository repo) path (.getTree commit))]
     (when tree-walk
-      (.getObjectId tree-walk 0))))
+      (.getObjectId ^org.eclipse.jgit.treewalk.TreeWalk tree-walk 0))))
```
Comment (edannenberg) [inline src/clj_jgit/porcelain.clj]: This hint doesn't seem to be required, at least I got no reflection warning leaving it out. If it is required it's probably better to put the hint into the let where the var is defined.

**Comment by edannenberg** [comment]
Wanted this in before I started with #54, also avail. as `0.9.1-SNAPSHOT` on Clojars.

**Comment by pmonks** [comment]
Apologies for the unresponsiveness - day job has sucked up all my time in recent months.

**Comment by edannenberg** [comment]
No worries and thank you for the PR.

**Pull request closed without merging**